e2e/tests/components/tag-spec.ts

```diff
@@ -0,0 +1,77 @@
+import { test, expect, devices } from '@playwright/test';
+
+test.describe('Tag', () => {
+
+  let componentUrls: string[] = [];
+  const workingDirectory = process.cwd();
+  // split the last part of the workingDirectory path and add the storybook path
+  const localStorybookPath = `file://${workingDirectory.split('/').slice(0, -1).join('/')}/packages/react/storybook-static/index.html?path=/story/`;
+  const componentPath = `${localStorybookPath}components-tag`;
+  console.log(componentPath);
+
+  [devices['Desktop Chrome'], devices['iPhone SE']].forEach(device => {
+    test(`tag ${device.isMobile ? 'mobile' : 'desktop'}`, async ({ page }) => {
+      page.setViewportSize(device.viewport);
+      await page.goto(componentPath);
+
+      if (componentUrls.length == 0) {
+        await expect(page.locator("#components-tag")).toBeVisible();
+        const components = await page.locator('[data-parent-id="components-tag"]').all();
+        // loop components and get each's url
+        for (const component of components) {
+          await component.getAttribute('href').then((href) => {
+            // don't add anything containing 'playground' to the list
+            if (href && !href.includes('playground')) {
+              componentUrls.push(href as string);
+            }
+          });
+        }
+      }
+    });
+
+    test(`subTest-${device.isMobile ? 'mobile' : 'desktop'}`, async ({ page }) => {
+      page.setViewportSize(device.viewport);
+      for (const componentUrl of componentUrls) {
+        await page.goto(`file://${componentUrl}`);
+        const iFrame = page.frameLocator('iframe[title="storybook-preview-iframe"]');
+        const container = iFrame.locator('body');
+
+        // name for the screenshot
+        const name = `${componentUrl.split('/').pop()}-${device.isMobile ? 'mobile' : 'desktop'}`;
+
+        // take screenshot, also waits for all to render
+        await expect(container).toHaveScreenshot(`${name}.png`);
+
+        const tags = await container.locator('[data-playwright="true"]').all();
+
+        for (const [index, tag] of tags.entries()) {
+          await tag.scrollIntoViewIfNeeded();
+          const tagBoundingBox = await tag.boundingBox() || { x: 0, y: 0, width: 0, height: 0 };
```

Comment:
When can the boundingBox be missing?

Reply:
TypeScript says so, so I guess it can return some null etc 🤷